src/MicrosoftTeams.ts

```diff
@@ -417,25 +443,147 @@ namespace microsoftTeams
+            width = Math.min(width, (currentWindow.outerWidth - 400));
+            height = Math.min(height, (currentWindow.outerHeight - 200));
+
             // Convert any relative URLs into absolute ones before sending them over to our parent window
             let link = document.createElement("a");
-            link.href = authenticateParameters.url;
+            link.href = url;
 
-            let messageId = sendMessage(
-                "authentication.authenticate",
-                link.href,
-                authenticateParameters.width,
-                authenticateParameters.height);
-            callbacks[messageId] = (success: boolean, response: string) =>
+            if (hostClientType === hostClientTypes.desktop)
             {
-                if (success)
+                // We are running in our desktop app so give our main process a heads up on the URL we are about
+                // to call window.open with; this will allow the window.open call to pass through without getting
+                // kicked out into the browser (which is the default behavior)
+                let messageId = sendMessageRequest(parentWindow, "authentication.allowWindowOpen", [ link.href ]);
+                callbacks[messageId] = (allowed: boolean) =>
                 {
-                    authenticateParameters.successCallback(response);
+                    if (allowed)
+                    {
+                        // Open the window with a desired set of electron BrowserWindow features
+                        childWindow = currentWindow.open(link.href, "_blank", "width=" + width + ", height=" + height);
+                    }
+                    else
+                    {
+                        throw new Error("Authentication is only supported for URLs matching the pattern registered in the manifest.");
+                    }
+                };
+            }
+            else
+            {
+                // We are running in the browser so we need to center the new window ourselves
+                let left: number = (typeof currentWindow.screenLeft !== "undefined") ? currentWindow.screenLeft : currentWindow.screenX;
+                let top: number = (typeof currentWindow.screenTop !== "undefined") ? currentWindow.screenTop : currentWindow.screenY;
+                left += (currentWindow.outerWidth / 2) - (width / 2);
+                top += (currentWindow.outerHeight / 2) - (height / 2);
+
+                // Open the window with a desired set of standard browser features
+                childWindow = currentWindow.open(link.href, "_blank", "toolbar=no, location=yes, status=no, menubar=no, top=" + top + ", left=" + left + ", width=" + width + ", height=" + height);
+            }
+
+            // Start monitoring the authentication window so that we can detect if it gets closed before the flow completes
+            startAuthenticationWindowMonitor();
+        }
```

Comment:
I think we'll actually want to avoid this if childWindow is null. Else, unless I didn't follow this correctly, you'll start the interval and in the first iteration notice that the window was closed and fail with "CancelledByUser".